pallets/rewards/src/types.rs

```diff
@@ -0,0 +1,296 @@
+use cfg_traits::ops::ensure::{
+	EnsureAdd, EnsureAddAssign, EnsureFixedPointNumber, EnsureInto, EnsureSub, EnsureSubAssign,
+};
+use frame_support::pallet_prelude::*;
+use sp_runtime::{traits::Zero, ArithmeticError, FixedPointNumber, FixedPointOperand};
+
+/// Type that contains the stake properties of stake class
+#[derive(Encode, Decode, TypeInfo, MaxEncodedLen, RuntimeDebug)]
+#[cfg_attr(test, derive(PartialEq))]
+pub struct CurrencyInfo<Balance, Rate, GroupId, MaxMovements: Get<u32>> {
+	pub group_id: Option<GroupId>,
```

Comment:
Why `Option<GroupId>`?

Reply:
You need to configure which GroupId will be used for a currency. A non-attached Currency initially has a `None` group.